src/Razor/src/Microsoft.AspNetCore.Razor.LanguageServer/Diagnostics/RazorTranslateDiagnosticsService.cs

```diff
@@ -134,27 +120,35 @@ private Diagnostic[] FilterCSharpDiagnostics(Diagnostic[] unmappedDiagnostics, R
     private Diagnostic[] MapDiagnostics(
         RazorLanguageKind languageKind,
         IReadOnlyList<Diagnostic> diagnostics,
+        IDocumentSnapshot documentSnapshot,
         RazorCodeDocument codeDocument,
         SourceText sourceText)
     {
-        if (languageKind != RazorLanguageKind.CSharp)
-        {
-            // All other non-C# requests map directly to where they are in the document.
-            return diagnostics.ToArray();
-        }
-
-        var mappedDiagnostics = new List<Diagnostic>();
+        var projects = RazorDiagnosticConverter.GetProjectInformation(documentSnapshot);
+        using var mappedDiagnostics = new PooledArrayBuilder<Diagnostic>();
 
         for (var i = 0; i < diagnostics.Count; i++)
         {
             var diagnostic = diagnostics[i];
 
-            if (!TryGetOriginalDiagnosticRange(diagnostic, codeDocument, sourceText, out var originalRange))
+            // C# requests don't map directly to where they are in the document.
+            if (languageKind == RazorLanguageKind.CSharp)
+            {
+                if (!TryGetOriginalDiagnosticRange(diagnostic, codeDocument, sourceText, out var originalRange))
+                {
+                    continue;
+                }
+
+                diagnostic.Range = originalRange;
+            }
+
+            if (diagnostic is VSDiagnostic vsDiagnostic)
             {
-                continue;
+                // We're the ones reporting the diagnostic, and it shows up as coming from our filename (not the generated one), so
+                // the project info should be consistent too
+                vsDiagnostic.Projects = projects;
```

Comment:
This change is just quality of life. We were taking the project name reported by the downstream server, but for Html it was blank, and for TypeScript it was "Miscellaneous". This change means we always report as our understanding of the project, which makes sense anyway since we've mapped the range and filename back to our understanding of the source file anyway.